include/ignition/gazebo/components/Model.hh

```diff
@@ -29,14 +32,64 @@ namespace gazebo
 {
 // Inline bracket to help doxygen filtering.
 inline namespace IGNITION_GAZEBO_VERSION_NAMESPACE {
+namespace serializers
+{
+  class SdfModelSerializer
+  {
+    /// \brief Serialization for `sdf::Model`.
+    /// \param[in] _out Output stream.
+    /// \param[in] _time Model to stream
+    /// \return The stream.
+    public: static std::ostream &Serialize(std::ostream &_out,
+                const sdf::Model &_model)
+    {
+      sdf::ElementPtr modelElem = _model.Element();
+      if (!modelElem)
+      {
+        ignerr << "Unable to serialize sdf::Model" << std::endl;
+        return _out;
+      }
+
+      _out << "<?xml version=\"1.0\" ?>"
+           << "<sdf version='" << modelElem->OriginalVersion() << "'>"
+           << modelElem->ToString("")
+           << "</sdf>";
+      return _out;
+    }
+
+    /// \brief Deserialization for `sdf::Model`.
+    /// \param[in] _in Input stream.
+    /// \param[out] _model Model to populate
+    /// \return The stream.
+    public: static std::istream &Deserialize(std::istream &_in,
+                sdf::Model &_model)
+    {
+      sdf::Root root;
+      std::string sdf(std::istreambuf_iterator<char>(_in), {});
+
+      sdf::Errors errors = root.LoadSdfString(sdf);
+      if (!errors.empty())
+      {
+        ignerr << "Unable to unserialize sdf::Model" << std::endl;
+        return _in;
+      }
+
+      _model.Load(root.Element()->GetElement("model"));
```

Comment:
I think doing `_model = *_root.Model()` should work and would avoid loading the model twice.

Reply:
Changed in de79f9a.